.pre-commit-config.yaml

```diff
@@ -20,7 +20,7 @@ repos:
 
       - id: cargo-test
         name: cargo test
-        entry: cargo nextest run --workspace --all-features --fail-fast --no-tests=pass
+        entry: cargo test --workspace --all-features
```

Comment:
I wonder if we could switch everything (local, pre-commit, and CI) to use nextest? It both runs faster and has easier to read reporting.

Reply:
That's a great idea - nextest does have better output and parallel execution. Would you like me to tackle that in this PR or create a follow-up issue? It would involve:
- Adding nextest to dev dependencies
- Updating pre-commit hooks
- Updating CI workflow
- Updating AGENTS.md documentation

I'm inclined to do it as a follow-up to keep this PR focused on multi-stream atomicity, but happy to do it now if you prefer.